arraycontext/container/traversal.py

```diff
@@ -732,13 +733,16 @@ def _unflatten(template_subary: ArrayOrContainerT) -> ArrayOrContainerT:
 
 # {{{ numpy conversion
 
-def from_numpy(ary: Any, actx: ArrayContext) -> Any:
+def from_numpy(
+        ary: Union[np.ndarray, np.generic, Number],
+        actx: ArrayContext) -> ArrayOrContainerT:
     """Convert all :mod:`numpy` arrays in the :class:`~arraycontext.ArrayContainer`
     to the base array type of :class:`~arraycontext.ArrayContext`.
 
     The conversion is done using :meth:`arraycontext.ArrayContext.from_numpy`.
     """
-    def _from_numpy_with_check(subary: Any) -> Any:
+    def _from_numpy_with_check(subary: Union[np.ndarray, np.generic, Number]) \
```

Comment:
Last time I checked `mypy` didn't like `Number` in the type hints
https://github.com/python/mypy/issues/3186
Doesn't seem like there's a good workaround besides making another union of all the number types we care about :\

Reply:
Thanks for taking a look! After briefly considering [this](https://posita.github.io/numerary/0.3/), I decided to go with the lower-fidelity :hankey: approach instead.

https://github.com/inducer/arraycontext/compare/7c2e18355cd0b2ee9861aa19d63bd508ecdc8ea0..88f88a6a774a3bf8bfb388925bf743d89e558fad

Reply:
Yeah, that `_ScalarLike` looks good to me!

It seems to be missing the changes in `traversal.py`?

Reply:
Whoops. Thanks for catching that! #140